main.js

```diff
@@ -8,9 +8,51 @@
         attribution: '&amp;copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
     }).addTo(map);
 
+    //Array of markers and a parallel array that holds their date of occurance
+    var markers = new Array();
+    var dates = new Array();
+
     //WPRDC data
     const WPRDC_BASE_URL = 'https://data.wprdc.org/api/action/datastore_search_sql?sql=';
 
+    //Create a new Date object for the current date
+    var currentDate = new Date();
+
+    //The following are functions that display records created within 1, 7, and
+    //30 days respectively
+    function displayToday() {
+        markers.forEach((marker, i) => {
+            if (dates[i][0] == currentDate.getFullYear() && dates[i][1] == currentDate.getMonth() + 1 && dates[i][2] == currentDate.getDate()) {
+                marker.addTo(map);
+            } else {
+                map.removeLayer(marker);
+            }
+        });
+    }
+
+    function displayPastWeek() {
+        markers.forEach((marker, i) => {
+            var tmpDate = new Date(dates[i][0], dates[i][1] - 1, dates[i][2]);
+            var diff = Math.ceil(Math.abs(currentDate.getTime() - tmpDate.getTime()) / (86400000));
+            if (diff <= 7) {
+                marker.addTo(map);
+            } else {
+                map.removeLayer(marker);
+            }
+        });
+    }
+
+    function displayPastMonth() {
+        markers.forEach((marker, i) => {
+                marker.addTo(map);
+        });
+    }
+
+    //Add listeners for each radio button
+    document.getElementById("radioDay").addEventListener("click", displayToday);
```

Comment:
Again, this works, and I think it would otherwise be preferred, but for our case it would be better to bind in the HTML